doc/RELEASE_NOTES.md

```diff
@@ -0,0 +1,828 @@
+
+Note that for transient tools, where data for the transient tool is collected locally on the host the tool is registered, the collected data is sent to the Tool Data Sink when the benchmark script deems it won't impact behavior of the benchmark itself.
+
+### Prometheus tools: `node-exporter` and `dcgm`
+
+Two new pbench "tools" have been added, `node-exporter` and `dcgm`.  If one registers either or both of these new tools (e.g. via `pbench-register-tools --name=node-exporter`), then the Tool Meister sub-system will run the `node_exporter` code on the registered hosts, and a local instance of Prometheus to collect the data.  The collected Prometheus data is stored in the pbench result directory as a tar ball at: `${pbench_run}/<script>_<config>_YYYY.MM.DDTHH.mm.ss/tools-<group>/prometheus`.
+
+For the duration of the run, the Prometheus instance is available on `localhost:9090` if one desires to review the metrics being collected live.
+
+_**NOTE WELL**_: like all the other "tools" the `pbench-agent` supports, the `node-exporter` and `dcgm` tools themselves need to be installed separately on the registered hosts.
+
+The new `dcgm` tool requires Python 2, an Nvidia based install which might conflict with the Pbench Agent's Python 3 operational requirement in some cases.
+
+### The PCP tool
+
+Just like the new Prometheus based tools, you can register "PCP" as a persistent tool using: `pbench-register-tool --name=pcp`.  This will cause each Tool Meister on the hosts for which PCP is registered to start a PMCD instance, and the Tool Data Sink will run `pmlogger` processes for each of hosts to collect the data at the requested interval.
+
+The PCP support also allows you to register PCP as a transient tool, where it is started and stopped around each benchmark invoked.  Use the name `pcp-transient` when registering (e.g. `pbench-register-tool --name=pcp-transient`).
+
+_**NOTE AS WELL**_: like all the other "tools" the `pbench-agent` supports, the `pcp` tools themselves need to be installed separately on the registered hosts.
+
+
+## CLI Interfaces to the Tool Meister Sub-system
+
+Two new CLI command are provided to start and stop the Tool Meister sub-system, `pbench-tool-meister-start` and `pbench-tool-meister-stop`.  These two interfaces are mostly used by the benchmark convenience scripts provided by the Pbench Agent.
+
+If you were using the `pbench-start-tools`, `pbench-stop-tools`, and `pbench-postprocess-tools` before, you must now invoke `pbench-tool-meister-start` before using the start/stop/postprocess tool interfaces.
+
+Further the `pbench-send-tools` CLI interface has been added to instruct the Tool Meister sub-system when it is time to send the collected data from transient tools to the Tool Data Sink.  The typical sequence is to "start", "stop", "send", "post-process".  Now you can "send" and "post-process" the transient tool data any time before `pbench-tool-meister-stop` is called.
```

Comment:
The second paragraph is confusing.  We don't draw a careful enough distinction between `pbench-stop-tools` and `pbench-tool-meister-stop`, and the shorthand used in depicting the "typical sequence" blurs things further.

Also, AFAIK there's no requirement to do the post-processing before the Tool Meister is stopped, so perhaps it would ease things to leave that to a separate paragraph (i.e., before doing postprocessing, we have to do a "send", and we have to do the "send" before stopping the TM, but the postprocessing can be done at any point once the data is available locally).

I think I would approach this by saying that using `pbench-start-tools` and `pbench-stop-tools` requires the TM to be running, so they must be done after invoking `pbench-tool-meister-start` and before invoking `pbench-tool-meister-stop`.  And, then I would go on to add that `pbench-send-tools` also depends on the TM, so it must be invoked before the TM is stopped.  And, then finally I would point out that postprocessing depends on having had the data sent from the TMs, and so it cannot be used until after `pbench-send-tools`.  And, then you could round it out with a comment that the sequence start-tools, stop-tools, send-tools can be performed repeatedly without stopping the TM.

Reply:
I hesitated to add the second paragraph.  I have always been against the `pbench-start-tools` and `pbench-stop-tools` being publicly supported interfaces.  And this is one of the reasons why.

I think it would be best to just remove this paragraph altogether.  To properly explain how to use the new `pbench-send-tools` invites the use of the interface.

Reply:
> To properly explain how to use the new pbench-send-tools invites the use of the interface.

Kind of like implementing a self-destruct button?  🤣

> I think it would be best to just remove this paragraph altogether.

Okie.